lib/vspc/handlers.go

```diff
@@ -0,0 +1,204 @@
+		h.handleVmotionAbort(w, tc, b)
+	}
+}
+
+// handleVMUUID handles the telnet vm-uuid response
+func (h *handler) handleVMUUID(w io.Writer, tc *telnet.Conn, b []byte) {
+	defer trace.End(trace.Begin("handling VMUUID"))
+	vmuuid := strings.Replace(string(b[3:len(b)-1]), " ", "", -1)
+	log.Infof("vmuuid of the connected containerVM: %s", vmuuid)
+	// check if there exists another vm with the same vmuuid
+	cvm, exists := h.vspc.cVM(vmuuid)
+	if !exists || !cvm.isInVMotion() {
+		// create a new vm associated with this telnet connection
+		cvm = newCVM(tc)
+		log.Infof("attempting to connect to the attach server")
+		remoteConn, err := net.Dial("tcp", fmt.Sprintf("%s:%d", h.vspc.attachSrvrAddr, h.vspc.attachSrvrPort))
+		if err != nil {
+			log.Errorf("cannot connect to the attach-server: %v", err)
+		}
+		cvm.remoteConn = remoteConn
+		cvm.vmUUID = vmuuid
+		h.vspc.addCVM(vmuuid, cvm)
+		// relay Reads from the remote system connection to the telnet connection associated with this vm
+		go h.vspc.relayReads(cvm, remoteConn)
+	} else { //the vm existed before and was shut down or vmotioned
+		log.Infof("established second serial-port telnet connection with vm (vmuuid: %s)", cvm.vmUUID)
+		cvm.Lock()
+		defer cvm.Unlock()
+		cvm.prevContainerConn = cvm.containerConn
+		cvm.containerConn = tc
+	}
+}
+
+// handleKnownSuboptions handles the known suboptions telnet command
+func (h *handler) handleKnownSuboptions(w io.Writer, b []byte) {
+	defer trace.End(trace.Begin("handling KNOWN-SUBOPTIONS"))
+	var resp []byte
+	suboptions := b[3 : len(b)-1]
+	resp = append(resp, []byte{telnet.Iac, telnet.Sb, VmwareExt, KnownSuboptions2}...)
+	resp = append(resp, suboptions...)
+	resp = append(resp, telnet.Iac, telnet.Se)
+	log.Debugf("response to KNOWN-SUBOPTIONS: %v", resp)
+
+	if bytes.IndexByte(suboptions, GetVMVCUUID) != -1 && bytes.IndexByte(suboptions, VMVCUUID) != -1 {
+		resp = append(resp, getVMUUID()...)
+	}
+	w.Write(resp)
+}
+
+// handleDoProxy handles the DO-PROXY telnet command
+func (h *handler) handleDoProxy(w io.Writer, b []byte) {
+	defer trace.End(trace.Begin("handling DO-PROXY"))
+	var resp []byte
+	resp = append(resp, []byte{telnet.Iac, telnet.Sb, VmwareExt, WillProxy, telnet.Iac, telnet.Se}...)
+	log.Debugf("response to DO-PROXY: %v", resp)
+	w.Write(resp)
+}
+
+// handleVmotionBegin handles the VMOTION-BEGIN telnet command
+func (h *handler) handleVmotionBegin(w io.Writer, tc *telnet.Conn, b []byte) {
+	defer trace.End(trace.Begin("handling VMOTION-BEGIN"))
+	if cvm, exists := h.vspc.cvmFromTelnetConn(tc); exists {
+		cvm.Lock()
+		cvm.inVmotion = true
+		cvm.Unlock()
+		ch := make(chan struct{})
+		cvm.vmotionStartedChan <- ch
+		<-ch
+	}
+	seq := b[3 : len(b)-1]
+	var escapedSeq []byte
+	for _, v := range seq {
+		if v == telnet.Iac {
+			escapedSeq = append(escapedSeq, telnet.Iac)
+		}
+		escapedSeq = append(escapedSeq, v)
+	}
+	secret := make([]byte, 4)
+	var escapedSecret []byte
+	rand.Read(secret)
```

Comment:
crypto/rand does not have/need a seed https://golang.org/pkg/crypto/rand/
math/rand does.